packages/components/src/components/hds/form/field/index.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { getElementId } from '../../../../utils/hds-get-element-id.js';
+import {
+  ariaDescribedBy,
+  registerAriaDescriptionElement,
+  unregisterAriaDescriptionElement,
+} from '../../../../utils/hds-aria-described-by.ts';
+import { HdsFormLayoutValues } from '../types.ts';
+import HdsFormLabelComponent from '../label/index.ts';
+import HdsFormHelperTextComponent from '../helper-text/index.ts';
+import HdsFormCharacterCountComponent from '../character-count/index.ts';
+import HdsFormErrorComponent from '../error/index.ts';
+
+import type { HdsFormLayouts } from '../types.ts';
+import type { ComponentLike, WithBoundArgs } from '@glint/template';
+import type { HdsYieldSignature } from '../../yield/index.ts';
+
+export const LAYOUT_TYPES = Object.values(HdsFormLayoutValues);
+
+export interface HdsFormFieldSignature {
+  Args: {
+    extraAriaDescribedBy?: string;
+    contextualClass?: string;
+    isOptional?: boolean;
+    isRequired?: boolean;
+    layout?: HdsFormLayouts;
+  };
+  Blocks: {
+    default: [
+      {
+        Label?: WithBoundArgs<
+          typeof HdsFormLabelComponent,
+          'contextualClass' | 'controlId' | 'isRequired' | 'isOptional'
+        >;
+        HelperText?: WithBoundArgs<
+          typeof HdsFormHelperTextComponent,
+          'contextualClass' | 'controlId' | 'onInsert'
+        >;
+        Control?: ComponentLike<HdsYieldSignature>;
+        CharacterCount?: WithBoundArgs<
+          typeof HdsFormCharacterCountComponent,
+          'contextualClass' | 'controlId' | 'onInsert'
+        >;
+        Error?: WithBoundArgs<
+          typeof HdsFormErrorComponent,
+          'contextualClass' | 'controlId' | 'onInsert' | 'onRemove'
+        >;
+        id?: string;
+        ariaDescribedBy?: string;
+      }
+    ];
+  };
+  Element: HTMLDivElement;
+}
+
+@ariaDescribedBy
+class HdsFormFieldComponent extends Component<HdsFormFieldSignature> {
```

Comment:
@fivetanley I started converting the Field and Fieldset components and I'm having trouble finding the right format in TypeScript to make it work with the class decorator. When time allows, would you be able to give it a look? 🙏